x/tokenomics/token_logic_module/types.go

```diff
@@ -60,28 +66,35 @@ func (tlm TokenLogicModuleId) EnumIndex() int {
 // actors if their stake falls below a certain threshold.
 type TokenLogicModule interface {
 	GetId() TokenLogicModuleId
-	Process(
-		context.Context,
-		cosmoslog.Logger,
-		*PendingSettlementResult,
-		*sharedtypes.Service,
-		*sessiontypes.SessionHeader,
-		*apptypes.Application,
-		*sharedtypes.Supplier,
-		cosmostypes.Coin, // This is the "actualSettlementCoin" rather than just the "claimCoin" because of how settlement functions; see ensureClaimAmountLimits for details.
-		*servicetypes.RelayMiningDifficulty,
-	) error
+	Process(context.Context, cosmoslog.Logger, TLMContext) error
+}
+
+// TLMUsedParams provides a field for each module Params type upon which any TLM depends.
```

Comment:
It's unclear to me why we need this.

1. #PUC Why
2. Do we really need it?

Reply:
I suppose the anticipation was that we may need to take other module params into consideration but perhaps that's not realistic.